veps/sig-network/passt-migration-proposal.md

```diff
@@ -0,0 +1,314 @@
+- The number of concurrent live migration per node is already limited and bound by the number of virt-handler worker goroutines.
+
+## Update/Rollback Compatibility
+
+- Live migration of VMs with `passt` network binding, from nodes running older versions of KubeVirt to an upgraded KubeVirt environment will result in TCP connections being reset. 
+The migration will succeed, but since the old instance of passt does not have any of the necessary code and infrastructure to properly freeze the socket, record the state and send the state to the target `passt` instance, it will receive nothing, and not attempt to restore the socket state. From the client perspective, the connection would be reset and a new handshake will be necessary.      
+- Rollback of passt migration functionality (in case system stability is impacted) will be possible by disabling of the feature gate. It is expected that passt would still work, but live migrations would disconnect TCP connections.
+
+## Functional Testing
+
+Existing e2e tests cover basic passt connectivity and remain sufficient with vhost-user support. Dedicated e2e tests for seamless migration will verify TCP connection persistence. Sticky IP addresses are required, so tests will use secondary interfaces with hardcoded IPs.
+
+## Implementation Phases
+
+1. Implement vhost-user in passt-sidecar.
+2. Implement calls to passt-repair in virt-handler.
+3. Enhance e2e tests and demonstrate migration without disconnection.
+
+Phases can proceed in parallel.
+
+## Feature Lifecycle Phases
+
+`passt` is currently a network binding plugin external to the KubeVirt core, and its use as a network binding is not subject to lifecycle phases. However, the calls to passt-repair from virt-handler impose a risk and should be protected by a feature gate and subject to lifecycle phases.
+
+### Alpha
+
+A feature-gate named `passtTCPMigration` will be introduced, requiring explicit user opt-in. virt-handler will not run `passt-repair` unless the feature gate is enabled.
+
+passt-sidecar will not condition the feature gate and will populate DomainXML in vhost-user mode regardless of the gate. This part of the code is external to KubeVirt core.
+
+### Beta
+
+For up to two releases, the live migration process using `passt` network binding, will be optimized. Specifically, `past-repair` and its integration points will be ironed out. 
+The criteria to move to from Alpha to Beta will be stability of the live migration process, and an agreement that the decision to separate `passt-repair` from `passt` was a sustainable one, as other alternatives do exist.
+While the feature gate is still in place for protection, the binding will be migrated to the core and will cease to be a plugin. 
+
+### GA
+
+GA can be declared when it is part of the core binding and a proper alternative to the masquerade binding.
+At this point, the migration will include seamless connectivity treatment by default with no option to disable it.
```

Comment:
Will the new passt core binding be applicable to users using network providers without sticky IPs?